includes/Media/Media.php

```diff
@@ -555,4 +583,43 @@ protected function is_poster( int $post_id ): bool {
 
 		return false;
 	}
+
+	/**
+	 * Filter the default value of is muted, to check to see if the video has audio track.
+	 *
+	 * @since 1.10.0
+	 *
+	 * @param mixed  $value     The value to return, either a single metadata value or an array
+	 *                          of values depending on the value of `$single`.
+	 * @param int    $object_id ID of the object metadata is for.
+	 * @param string $meta_key  Metadata key.
+	 * @param bool   $single    Whether to return only the first value of the specified `$meta_key`.
+	 *
+	 * @return bool|bool[]
+	 */
+	public function filter_default_value_is_muted( $value, $object_id, $meta_key, $single ) {
+		if ( self::IS_MUTED_POST_META_KEY !== $meta_key ) {
+			return $value;
+		}
+		$mime_types = $this->get_allowed_mime_types();
+		if ( ! is_array( $mime_types ) || ! isset( $mime_types['video'] ) ) {
+			return $value;
+		}
+
+		$video_mime_types  = $mime_types['video'];
+		$current_mime_type = get_post_mime_type( $object_id );
+
+		if ( ! in_array( $current_mime_type, $video_mime_types, true ) ) {
+			return $value;
+		}
+
+		$meta_data = wp_get_attachment_metadata( $object_id );
+		if ( ! $meta_data ) {
+			return $value;
+		}
+
+		$is_muted = ! isset( $meta_data['audio'] );
```

Comment:
Checking if already tracks exists, works on all mp4/webm I have tested on. It doesn't work on mpgs, but that isn't an issue here.

Reply:
Worst case audio detection fails and we mark a video as muted even though it isn't. But I suppose that's acceptable.